src/transformers/pipelines/object_detection.py

```diff
@@ -0,0 +1,170 @@
+        )
+
+    def __call__(
+        self,
+        images: Union[str, List[str], "Image", List["Image"]],
+        threshold: Optional[float] = 0.9,
+    ) -> Union[Predictions, List[Prediction]]:
+        """
+        Detect objects (bounding boxes & classes) in the image(s) passed as inputs.
+
+        Args:
+            images (:obj:`str`, :obj:`List[str]`, :obj:`PIL.Image` or :obj:`List[PIL.Image]`):
+                The pipeline handles three types of images:
+
+                - A string containing a http link pointing to an image
+                - A string containing a local path to an image
+                - An image loaded in PIL directly
+
+                The pipeline accepts either a single image or a batch of images, which must then be passed as a string.
+                Images in a batch must all be in the same format: all as http links, all as local paths, or all as PIL
+                images.
+            threshold (:obj:`float`, `optional`, defaults to 0.9):
+                The probability necessary to make a prediction.
+
+        Return:
+            A list of dictionaries or a list of list of dictionaries containing the result. If the input is a single
+            image, will return a list of dictionaries, if the input is a list of several images, will return a list of
+            list of dictionaries corresponding to each image.
+
+            The dictionaries contain the following keys:
+
+            - **label** (:obj:`str`) -- The label identified by the model.
+            - **score** (:obj:`float`) -- The score attributed by the model for that label.
+            - **box** (:obj:`List[Dict[str, int]]`) -- The bounding box of detected object in image's original size.
+        """
+        is_batched = isinstance(images, list)
+
+        if not is_batched:
+            images = [images]
+
+        images = [self.load_image(image) for image in images]
+
+        with torch.no_grad():
+            inputs = self.feature_extractor(images=images, return_tensors="pt")
+            outputs = self.model(**inputs)
+
+            target_sizes = torch.IntTensor([[im.height, im.width] for im in images])
+            annotations = self.feature_extractor.post_process(outputs, target_sizes)
+            for i, annotation in enumerate(annotations):
+                keep = annotation["scores"] > threshold
+                scores = annotation["scores"][keep]
+                labels = annotation["labels"][keep]
+                boxes = annotation["boxes"][keep]
+
+                annotation["scores"] = scores.tolist()
+                annotation["labels"] = [self.model.config.id2label[label.item()] for label in labels]
+                annotation["boxes"] = [self._get_clockwise_vertices(box) for box in boxes]
+
+                # {"scores": [...], ...} --> [{"score":x, ...}, ...]
+                keys = ["score", "label", "box"]
+                annotation = [dict(zip(keys, vals)) for vals in zip(*annotation.values())]
```

Comment:
You cannot do this this way.
Dict in python are not of deterministic order so `annotation.values()` is not guaranteed to be in the correct order ('score', 'label', 'box').
Even worse, this dependency is not explicity for the reader (so anyone might want to reorder the `keys` list and incur a bug.

```python
n = len(annotations["scores"]) # We're hopeful that annotations are actually of sane outputs with the same length of every key, that's very hopeful.

```python
annotation = [{"score": raw_annotation["score"][i], "label": raw_annotation["label"][i], "box": raw_annotation["box"][i] for i in range(n)]
```
I would argue that the real culprit is the `post_process` tbh, because the output might be *unsound* (different shapes for annotations is possible). As much as possible, we want to use struct/data formats that cannot allow any discrepancy to even exist (list of Annotation, won't allow any different length, which woudl result in erroneous attribution)

Let's see if we can move this logic directly within `post_process`. It seems much more natural in terms of responsability but again, I don't know who else uses `post_process` out there, so my assumptions might be wrong.

Reply:
I'll incorporate the changes 👍 
I've banked on the fact that since py3.6 dicts are insertion order preserving [here](https://docs.python.org/3/whatsnew/3.6.html#whatsnew36-compactdict). But even the doc says: 
> The order-preserving aspect of this new implementation is considered an implementation detail and **should not be relied upon**